.github/workflows/milestone-checker.yml

```diff
@@ -27,7 +27,7 @@ jobs:
         with:
           repository: "grafana/grafana-github-actions"
           path: ./actions
-          ref: be89ad434792280ebaa4d982ac72ba548b6f7095
+          ref: 3e07b3b88b3984f28067a76dce5a052f0c5d3c73
```

Comment:
This isn't sufficient. That action still hasn't been updated to use Node20. I've posted an issue in that repo, but not sure if there are alternate paths here.

Reply:
TBH the milestone checker has been broken for a while, (it never updates once a milestone has been set for me), I think we can retire it.

Reply:
cc: @jefferai @tmessi Any opinions?

Reply:
cc: @jefferai @tmessi Any opinions?

Reply:
Can we investigate why it isn't working? Perhaps it needs additional `types` of events so that it triggers?

Reply:
Ok. I think I addressed both things
- I borrowed an implementation from vault: https://github.com/hashicorp/vault/blob/main/.github/workflows/milestone-checker.yml
- Triggers have been updated to include `milestoned` and `demilestoned`. We needed to update the version of `actionlint` to pass a linter error -- it would previously complain that those trigger types didn't exist (https://github.com/rhysd/actionlint/releases/tag/v1.6.27)
- Vault's implementation does not require a third party action, so we don't have to worry about the node version they are using anymore